### ISSUE TITLE
Add EfficientNet Model - PyTorch

### ISSUE THREAD

**Issue by tanaymeh** [huggingface/transformers#15759]
# 🌟 New model addition

I would like to add the EfficientNet model to the Transformers library

## Model description
EfficientNet was proposed in the paper, **[EfficientNet: Rethinking Model Scaling for Convolutional Neural Networks](https://arxiv.org/abs/1905.11946)**.

EfficientNet is a convolutional neural network architecture and scaling method that uniformly scales all dimensions of depth/width/resolution using a compound coefficient. Unlike conventional practice that arbitrary scales these factors, the EfficientNet scaling method uniformly scales network width, depth, and resolution with a set of fixed scaling coefficients.

EfficientNets also transfer well and achieve state-of-the-art accuracy on CIFAR-100 (91.7%), Flowers (98.8%), and 3 other transfer learning datasets, with an order of magnitude fewer parameters.

## Open source status

* [x] the model implementation is available: [EfficientNet-PyTorch](https://github.com/lukemelas/EfficientNet-PyTorch)
* [x] the model weights are available: [PyTorch weights](https://github.com/lukemelas/EfficientNet-PyTorch/releases/tag/1.0)
* [x] who are the authors: Mingxing Tan, Quoc V. Le

**Comment by LysandreJik** [comment]
cc @NielsRogge

**Comment by IMvision12** [comment]
@LysandreJik is anyone working on this?? if not can i work on this

**Comment by heytanay** [comment]
Hi @IMvision12, because of existing commitments, I am no longer working on this.

If you want, you can take it up :)

**Comment by IMvision12** [comment]
@NielsRoggeWill it be worthwhile to include this model?
